File: backend/app/agents/critic_agent.py
from langchain_core.prompts import ChatPromptTemplate
from langchain_core.output_parsers import StrOutputParser
from app.core.llm import get_llm

llm = get_llm()

critic_prompt = ChatPromptTemplate.from_messages([
    ("system", "You are a strict research reviewer."),
    ("human", """Review this report:

{report}

Give:
- Score (out of 10)
- Strengths
- Weaknesses
- Final verdict
"""),
])

critic_chain = critic_prompt | llm | StrOutputParser()

File: syncai/src/lib/agents/criticAgent.ts
import { getModel } from "../llm";

export async function runCriticAgent(report: string): Promise<string> {
  const model = getModel();
  const result = await model.generateContent(
    `You are a strict research reviewer. Review the following research report and provide detailed feedback in Markdown format.

Report:
${report}

Provide your review with:

## Score
**X / 10**

## Strengths
- (bullet points)

## Weaknesses
- (bullet points)

## Verdict
(1–2 sentences summary)`,
  );

  return result.response.text();
}
